src/Config.php

```diff
@@ -87,6 +89,10 @@ function forget(array &$array, string $key): array
  */
 function override(...$arrays): array
```

Comment:
Would it not have been nicer to change this to

```
function override(array $baseArray, ...$arrays): array
```
?


Then you get rid of the `array_shift` below because we've already got the `$baseArray` as a simple required argument, and `$arrays` will always be a valid value to pass into the `foreach` (if called with only one argument, then `$arrays` is an empty array and the function just returns the untouched `$baseArray`).